test/test_data/success.rb:14: Expected `Typed::Result[Integer, String]` but found `T.any(Typed::Success[String], Typed::Failure[String])` for method result type https://srb.help/7005
    14 |  end
          ^^^
  Expected `Typed::Result[Integer, String]` for result type of method `test_generic_initializer`:
    test/test_data/success.rb:8:
     8 |  def test_generic_initializer(should_succeed)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Got `T.any(Typed::Success[String], Typed::Failure[String])` originating from:
    test/test_data/success.rb:10:
    10 |      Typed::Success.new("1")
              ^^^^^^^^^^^^^^^^^^^^^^^
    test/test_data/success.rb:12:
    12 |      Typed::Failure.new("")
              ^^^^^^^^^^^^^^^^^^^^^^

test/test_data/success.rb:26: Expected `Integer` but found `String("success")` for argument `payload` https://srb.help/7002
    26 |    Typed::Success[Integer].new("success")
                                        ^^^^^^^^^
  Expected `Integer` for argument `payload` of method `Typed::Success#initialize`:
    ./lib/typed/success.rb:30:
    30 |    sig { params(payload: Payload).void }
                         ^^^^^^^
  Got `String("success")` originating from:
    test/test_data/success.rb:26:
    26 |    Typed::Success[Integer].new("success")
                                        ^^^^^^^^^
